src/bun.js/api/server.zig

```diff
@@ -1586,14 +1588,16 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
                 this.request_body = null;
             }
 
-            const server = this.server;
-            server.request_pool_allocator.put(this);
-            server.onRequestComplete();
+            if(this.server) |server| {
+                this.server = null;
+                server.request_pool_allocator.put(this);
```

Comment:
maybe we shouldn't make server nullable, since it means we don't have a good mechanism for freeing it since the allocator is server-relative.